### PR TITLE
Change validator to respect local/remote inference.

### PR DESCRIPTION
**Pull request by JosephCatrambone**
Also only push to Pypi on release tag.

### CODE REVIEW

File: validator/main.py [hunk 1]
```diff
@@ -289,3 +293,31 @@ def validate(
...
+            "inputs": [
+                {
+                    "name": "message",
+                    "shape": [len(model_input)],
+                    "data": model_input,
+                    "datatype": "BYTES"
+                }
+            ]
+        }
+        response = self._hub_inference_request(
```
Comment (AlejandroEsquivel) [inline validator/main.py:315]: from looking at the inference spec it seems that the response schema is:
`{ "score": float }` and doesn't contain `outputs`

**Review by AlejandroEsquivel — COMMENTED** [review]
the predict jailbreak method has a return signature of `    ) -> Union[List[float], List[dict]]:` is this being handled properly in `validate`?

File: validator/main.py [hunk 2]
```diff
@@ -65,8 +66,9 @@ def __init__(
             device: str = "cpu",
             on_fail: Optional[Callable] = None,
             model_path_override: str = "",
+            **kwargs,
```
Comment (AlejandroEsquivel) [inline validator/main.py:69]: I think we may want to use `self.use_local` to conditionally load the models

**Review by AlejandroEsquivel — CHANGES_REQUESTED** [review]

File: validator/main.py (hunk 1 above)
Comment (JosephCatrambone) [reply validator/main.py:315]: Does outputs get generated automatically from the endpoints?  Like "inputs"?

Comment (JosephCatrambone) [reply validator/main.py:315]: Fixed.

Comment (AlejandroEsquivel) [reply validator/main.py:315]: ah nope, the request body sent here should correspond exactly with InputRequest

File: validator/main.py (hunk 2 above)
Comment (JosephCatrambone) [reply validator/main.py:69]: Done.

File: validator/main.py [hunk 3]
```diff
@@ -289,3 +299,30 @@ def validate(
...
+        # This needs to be kept in-sync with app_inference_spec.
+        request_body = {
+            "inputs": [  # Required for legacy reasons.
+                {
+                    "name": "prompts",
+                    "shape": [len(model_input)],
+                    "data": model_input,
+                    "datatype": "BYTES"
+                }
+            ]
```
Comment (AlejandroEsquivel) [inline validator/main.py]: i.e this can just be `message`

Comment (AlejandroEsquivel) [reply validator/main.py]: didn't catch this earlier

File: validator/main.py (hunk 1 above)
Comment (JosephCatrambone) [reply validator/main.py:315]: Oh, wait.  So no "inputs" on the call to the remote?  Just `request_body = {"prompts": [str, str, str]}`?

Comment (AlejandroEsquivel) [reply validator/main.py:315]: thats right

File: validator/main.py (hunk 3 above)
Comment (AlejandroEsquivel) [reply validator/main.py]: sorry *prompts

**Review by AlejandroEsquivel — APPROVED** [review]
looks good to me 👌🏼

**Pull request merged**